### PR TITLE
DM-41711: Upgrade QuantumGraphExecutionReport to handle multiple overlapping graphs

### PR DESCRIPTION
**Pull request by eigerx**
Depends on lsst/pipe_base#426

## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cli/script/report.py [hunk 1]
```diff
@@ -106,16 +107,200 @@ def report(
...
+    their visible/shadowed states. Analyze one or more attempts at the same
+    processing on the same dataquery-identified "group" and resolve recoveries
+    and persistent failures. Identify mismatch errors between groups.
+
+    Parameters
+    ----------
+    butler_config : `str`
+            The Butler used for this report. This should match the Butler used
+            for the run associated with the executed quantum graph.
+    qgraph_uris : `Sequence[str]`
```
Comment (MichelleGower) [inline python/lsst/ctrl/mpexec/cli/script/report.py:139]: I think the formatting of this is different, but cannot find exact example.  See https://developer.lsst.io/python/numpydoc.html#sequence-types for list example.

File: tests/test_cliCmdReport.py [hunk 2]
```diff
@@ -74,19 +75,24 @@ def test_report(self):
...
         # Check that we can read from the command line
         self.assertEqual(result.exit_code, 0, clickResultMsg(result))
 
         # Check that we can open and read the file produced by make_reports
         with open(test_filename) as f:
             report_output_dict = yaml.load(f, Loader=SafeLoader)
+
+        with open("delete_me.yaml", "w") as f:
+            yaml.safe_dump(report_output_dict, f)
+
```
Comment (MichelleGower) [inline tests/test_cliCmdReport.py:88]: Debugging lines accidentally left in.

**Review by MichelleGower — APPROVED** [review]
Mostly suggestions about comments, but some code questions.  Merge approved.

File: python/lsst/ctrl/mpexec/cli/cmd/commands.py [hunk 3]
```diff
@@ -340,25 +340,73 @@ def update_graph_run(
...
-@ctrlMpExecOpts.qgraph_argument()
-@click.option("--full-output-filename", default="", help="Summarize report in a yaml file")
+@click.argument("qgraphs", nargs=-1)
+@click.option(
+    "--collections",
+    default=None,
+    help="Collection(s) associated with said graphs/processing."
+    " For use in `lsst.daf.butler.registry.queryDatasets` if paring down the query would be useful.",
+)
+@click.option("--where", default="", help="A 'where' string to use to constrain the collections, if passed.")
```
[the remark this answers is not in the document]
Comment (timj) [reply python/lsst/ctrl/mpexec/cli/cmd/commands.py]: You can override the default help if you use a standard option.

File: python/lsst/ctrl/mpexec/cli/cmd/commands.py [hunk 4]
```diff
@@ -341,25 +343,73 @@ def update_graph_run(
...
+    Butler `collections` and `where` options are for use in
+    `lsst.daf.butler.queryDatasets` if paring down the collections would be
+    useful. By default the collections and query be taken from the graphs.
 
     REPO is the location of the butler/registry config file.
 
-    QGRAPH is the URL to a serialized Quantum Graph file.
+    QGRAPHS is a `Sequence` of links to serialized Quantum Graphs which have
+    been executed and are to be analyzed.
     """
```
[the remark this answers is not in the document]
Comment (timj) [reply python/lsst/ctrl/mpexec/cli/cmd/commands.py:408]: No because click formats this string as help text and doesn't understand numpydoc.

**Pull request merged**